src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -51,29 +53,31 @@ public void printBasket() {
     }
 
     public boolean checkProductInBasket(String nameProduct) {
-        for (Product basketProduct : products) {
-            if (basketProduct.getName().equals(nameProduct)) {
-                return true;
-            }
-        }
-        return false;
+        return basket.containsKey(nameProduct);
     }
 
     public void cleanBasket() {
-        products.clear();
+        basket.clear();
     }
 
     public List<Product> removeByMame(String nameProduct) {
-        List<Product> deleteProducts = new LinkedList<>();
-        Iterator<Product> iterator = products.iterator();
-        while (iterator.hasNext()) {
-            Product basketProduct = iterator.next();
-            if (basketProduct.getName().contains(nameProduct)) {
-                deleteProducts.add(basketProduct);
-                iterator.remove();
-            }
+        if (basket.containsKey(nameProduct)) {
+            return basket.remove(nameProduct);
+        } else {
+            return new LinkedList<>();
```

Comment:
лучше вернуть пустой список так, по памяти лучше
```suggestion
            return Collections.emptyList();
```